### PR TITLE
Rds cluster pending maintenance list filter

### PR DESCRIPTION
**Pull request by sgungi-intuit**
Added filter to list pending maintenances for RDS Clusters

### CODE REVIEW

File: tests/test_rdscluster.py [hunk 1]
```diff
@@ -736,6 +736,25 @@ def test_clear_permissions(self):
         )
         self.assertEqual(len(restore_permissions_after), 0)
 
+    def test_pending_maintenance(self):
+        session_factory = self.record_flight_data("test_rdscluster_pending_maintenance")
```
Comment (StevenGunn) [inline tests/test_rdscluster.py:740]: @sgungi-intuit looks like tests are failing

This line should be verified locally then updated to "replay_flight_data"

File: tests/test_rdscluster.py [hunk 2]
```diff
@@ -736,6 +736,25 @@ def test_clear_permissions(self):
...
+                        "type": "pending-maintenance"
+                    }
+                ],
+            },
+            config={"region": "us-west-2"},
+            session_factory=session_factory,
+        )
+
+        resources = p.run()
+        self.assertEqual(len(resources), 0)
```
Comment (StevenGunn) [inline tests/test_rdscluster.py:756]: @sgungi-intuit shouldn't this match something?

Comment (StevenGunn) [reply tests/test_rdscluster.py:756]: for example, why are we not seeing a match for `docdb-2022-01-13-21-04-08`?

Comment (StevenGunn) [reply tests/test_rdscluster.py:756]: @sgungi-intuit I see, as there is pending maintenance for the DBs but not the cluster.... maybe best to have a test case that is a able to match when a cluster does have pending maintenance?

Comment (StevenGunn) [reply tests/test_rdscluster.py:756]: @sgungi-intuit I see- only the DB instances have pending maintenance, but not the RDS clusters themselves.

Probably best to have a test case with at least one cluster that does have pending maintenance to ensure the filter works as expected when there are matching resources

Comment (sgungi-intuit) [reply tests/test_rdscluster.py]: fixed the test case and added logic to pick arn instead of cluster id.

**Comment by sgungi-intuit** [comment]
Hi @ajkerrigan @thisisshi @kapilt can you take look at this PR when you get a chance, thanks.

**Comment by kapilt** [comment]
its currently failing ci due to a docs issue, in the example. 

```

/home/runner/work/cloud-custodian/cloud-custodian/docs/source/aws/resources/rds-cluster.rst:477: ERROR: Unexpected indentation.
/home/runner/work/cloud-custodian/cloud-custodian/docs/source/aws/resources/rds-cluster.rst:481: ERROR: Unexpected indentation.
```

**Comment by sgungi-intuit** [comment]
Hi @kapilt, all tested are passed now.

**Review by kapilt — APPROVED** [review]
lgtm, thanks

**Comment by sgungi-intuit** [comment]
Thanks @kapilt.

**Pull request merged**